lib/fastlane/plugin/wpmreleasetoolkit/helper/android/android_localize_helper.rb

```diff
@@ -171,6 +171,79 @@ def self.verify_pr_diff(main, library, main_strings, lib_strings, source_diff)
             end
           end
         end
+
+        ########
+        # @group Downloading translations from GlotPress
+        ########
+
+        # Create the `available_languages.xml` file.
+        #
+        # @param [String] res_dir The relative path to the `…/src/main/res` directory
+        # @param [Array<String>] locale_codes The list of locale codes to include in the available_languages.xml file
+        #
+        def self.create_available_languages_file(res_dir:, locale_codes:)
+          doc = Nokogiri::XML::Builder.new(encoding: 'UTF-8') do |xml|
+            xml.comment('Warning: Auto-generated file, do not edit.')
+            xml.resources do
+              xml.send(:'string-array', name: 'available_languages', translatable: 'false') do
+                locale_codes.each { |code| xml.item(code.gsub('-r', '_')) }
+              end
+            end
+          end
+          File.write(File.join(res_dir, 'values', 'available_languages.xml'), doc.to_xml)
+        end
+
+        # Perform some quick basic checks about an individual `<string>` tag and print warnings accordingly
+        #
+        # @param [Nokogiri::XML::Node] string_tag The XML tag/node to check
+        #
+        def self.quick_lint(string_tag)
+          if string_tag['formatted'] == 'false' && string_tag.content.include?('%%')
+            UI.important "Warning: translation for '#{string_tag['name']}' has attribute formatted=false, but still contains escaped '%%' in translation."
+          end
+        end
+        private_class_method :quick_lint
+
+        # Download translations from GlotPress
+        #
+        # @param [String] res_dir The relative path to the `…/src/main/res` directory
+        # @param [String] glotpress_project_url The base URL to the glotpress project to download the strings from
+        # @param [Array<Hash{Symbol=>String}>] locales_map An array of locales to download. Each item in the array must be a
+        #                                      Hash with keys `:glotpress` and `:android` containing the respective locale codes.
+        #
+        def self.download_from_glotpress(res_dir:, glotpress_project_url:, locales_map:)
+          en_file = File.join(res_dir, 'values', 'strings.xml')
+          en_xml = File.open(en_file) { |f| Nokogiri::XML(f, nil, Encoding::UTF_8.to_s) }
+          exclude_attrs = %w[name content_override] # Attributes that we don't want to replicate into translated XMLs
+          orig_attributes = en_xml.xpath('//string').map { |tag| [tag['name'], tag.attributes.reject { |k, _| exclude_attrs.include?(k) }] }.to_h
```

Comment:
Reading the XML attributes associated with each string key from the `values/strings.xml`, so that we can later replicate them in the other `strings.xml` downloaded by GlotPress.

 - We exclude `name` because obviously we don't need to replicate that one
 - We exclude `content_override` because it's a private attribute that only our tooling uses (to generate the original file when merging with lib strings), so isn't really useful
 - I've pondered about doing an allowlist instead of an exclusion list for the attributes to copy. wdyt?

Reply:
> I've pondered about doing an allowlist instead of an exclusion list for the attributes to copy. wdyt?

I think it's unlikely to change, but this would be the safer approach for sure, so I'd say it's worthwhile

Reply:
Fixed in b0e155af23eb84e6d7fc10e55965ea834e37267d